src/mlpack/methods/ann/ffn.hpp

```diff
@@ -160,12 +160,38 @@ class FFN
   template <class LayerType, class... Args>
   void Add(Args... args) { network.push_back(new LayerType(args...)); }
 
+  template <class LayerType, class... Args>
+  void AddNamed(std::string name, Args... args)
+  { 
+    network.push_back(new LayerType(args...));
+    // should raise error if duplicate
+    dictionary.insert(std::make_pair(name, new LayerType(args...)));
+  }
+
   /*
    * Add a new module to the model.
    *
    * @param layer The Layer to be added to the model.
    */
-  void Add(LayerTypes layer) { network.push_back(layer); }
+  void Add(LayerTypes layer, std::string name="")
+  { 
+    network.push_back(layer);
+    if(std::is_empty<std::string>(name))
+      dictionary["Layer" + network.size()] = layer;
+    else
+    {
+      // should raise error if duplicate
+      dictionary.insert(std::make_pair(name, layer));
+    }
+
+  }
+
+  std::vector<LayerTypes> Model(){return network;}
+
+  LayerTypes& getByName(std::string name)
```

Comment:
This doesn't follow the mlpack naming conventions; please take a look at the style guide.